src/modules/song/thunks.ts

```diff
@@ -53,6 +51,28 @@ export const uploadSong = createAsyncThunk(
 
     if ("error" in songResp) return;
 
+    const { songId } = songResp.data;
+
+    // get signed upload url for AWS
+    const audioUploadUrlResp = await thunkApi.dispatch(
+      songApi.endpoints.getAudioUploadUrl.initiate({
+        songId,
+        fileName: body.audio.name,
+      })
+    );
+
+    if ("error" in audioUploadUrlResp) return;
+
+    const { uploadUrl } = audioUploadUrlResp.data;
+    const audioBinaryStr = await getFileBinary(body.audio);
+
+    // upload audio to AWS, song audioUrl will be updated after it's transcoded
+    await fetch(uploadUrl, {
```

Comment:
Curious as to why we aren't adding this endpoint to song/api.ts and using thunkApi.dispatch here. Is it because the uploadUrl is a AWS url not a NEWM url?